File: microservices/card-service/src/main/java/hps/ma/dto/UpdatePhysicalSecurityOptionRequest.java
package hps.ma.dto;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class UpdatePhysicalSecurityOptionRequest {
    private Long cardId;
    private Boolean contactlessEnabled;
    private Boolean ecommerceEnabled;
    private Boolean tpeEnabled;
    private Boolean internationalWithdrawEnabled;
}

File: frontend/lib/dto/card_dto/UpdatePhysicalSecurityOptionRequest.dart
class UpdatePhysicalSecurityOptionRequest {
  final int cardId;
  final bool? contactlessEnabled;
  final bool? ecommerceEnabled;
  final bool? tpeEnabled;
  final bool? internationalWithdrawEnabled;

  UpdatePhysicalSecurityOptionRequest({
    required this.cardId,
    this.contactlessEnabled,
    this.ecommerceEnabled,
    this.tpeEnabled,
    this.internationalWithdrawEnabled,
  });

  Map<String, dynamic> toJson() {
    return {
      "cardId": cardId,
      if (contactlessEnabled != null) "contactlessEnabled": contactlessEnabled,
      if (ecommerceEnabled != null) "ecommerceEnabled": ecommerceEnabled,
      if (tpeEnabled != null) "tpeEnabled": tpeEnabled,
      if (internationalWithdrawEnabled != null) "internationalWithdrawEnabled": internationalWithdrawEnabled,
    };
  }
}
